File: 30days/04/04.rb
#!/usr/bin/env ruby

class Person
    attr_accessor :age
    def initialize(initialAge)
        if initialAge < 0
            puts "Age is not valid, setting age to 0."
            self.age = 0
        else
            self.age = initialAge
        end
    end

    def amIOld
        if self.age < 13
            puts "You are young."
        elsif (13..17).cover? self.age
            puts "You are a teenager."
        else
            puts "You are old."
        end
    end

    def yearPasses
        self.age = self.age + 1
    end
end

T = gets.to_i
(1..T).each? do
    age = gets.to_i
    p = Person.new(age)
    p.amIOld
    (1..3).each? do
        p.yearPasses
    end
    p.amIOld
    puts ""
end

File: 30days/04/04.php
<?php

class Person {
    public $age;

    public function __construct($initialAge) {
        if ($initialAge < 0) {
            echo "Age is not valid, setting age to 0.\n";
            $this->age = 0;
        }
        else {
            $this->age = $initialAge;
        }
    }

    public function amIOld() {
        if ($this->age < 13) {
            echo "You are young.\n";
        } elseif ($this->age >= 13 && $this->age < 18) {
            echo "You are a teenager.\n";
        } else {
            echo "You are old.\n";
        }
    }

    public function yearPasses() {
        $this->age += 1;
    }
}

$T = intval(fgets(STDIN));
 for($i=0;$i<$T;$i++){
     $age=intval(fgets(STDIN));
     $p=new Person($age);
     $p->amIOld();
     for($j=0;$j<3;$j++){
         $p->yearPasses();
     }
     $p->amIOld();
     echo "\n";
 }
?>
